lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_check_beta_deps.rb

```diff
@@ -1,31 +1,56 @@
+require 'yaml'
+
 module Fastlane
   module Actions
     class IosCheckBetaDepsAction < Action
       def self.run(params)
         require_relative '../../helper/ios/ios_version_helper'
         require_relative '../../helper/ios/ios_git_helper'
 
-        beta_pods = []
-        File.open(params[:podfile]).each do |li|
-          beta_pods << li if li.match('^\s*\t*pod.*beta')
+        yaml = YAML.load_file(params[:lockfile])
+        non_stable_pods = {} # Key will be pod name, value will be reason for flagging
+
+        # Find pods referenced by commit and branch to a repo
+        yaml['EXTERNAL SOURCES'].each do |pod, options|
+          non_stable_pods[pod] = 'commit' if params[:report_commits] && options.key?(:commit)
+          non_stable_pods[pod] = 'branch' if params[:report_branches] && options.key?(:branch)
         end
 
-        if beta_pods.count.zero?
-          UI.message('No beta pods found. You can continue with the code freeze.')
+        # Find pods whose resolved version matches the regex
+        version_pattern = params[:report_version_pattern]
+        unless version_pattern.nil? || version_pattern.empty?
+          regex = begin
+            Regexp.new(version_pattern)
+          rescue RegexpError
+            UI.user_error!("Invalid regex pattern: `#{version_pattern}`")
+          end
+          resolved_pods = yaml['PODS'].flat_map { |entry| entry.is_a?(Hash) ? entry.keys : entry }
+          resolved_pods.each do |line|
+            (pod, version) = /(.*) \((.*)\)/.match(line)&.captures
+            non_stable_pods[pod] = regex.source if regex.match?(version)
+          end
+        end
+
+        message = ''
+        if non_stable_pods.empty?
+          message << ALL_PODS_STABLE_MESSAGE
         else
-          message = "The following pods are still in beta:\n"
-          beta_pods.each do |bpod|
-            message << "#{bpod}\n"
+          message << NON_STABLE_PODS_MESSAGE
+          non_stable_pods.sort.each do |pod, reason|
+            message << " - #{pod} (currently points to #{reason})\n"
           end
-          message << 'Please update to the released version before continuing with the code freeze.'
         end
         UI.important(message)
+        { message: message, pods: non_stable_pods }
       end
 
       #####################################################
       # @!group Documentation
       #####################################################
 
+      ALL_PODS_STABLE_MESSAGE = 'All pods are pointing to a stable version. You can continue with the code freeze.'.freeze
+      NON_STABLE_PODS_MESSAGE = "Please create a new stable version of those pods and update the Podfile to the newly released version before continuing with the code freeze:\n".freeze
```

Comment:
Why declare these constants here instead of at the top of the `run` method?

Reply:
We need those constant to be accessible from outside the code of `def self.run` because they are also referenced by our unit tests 😉

Reply:
While we couldn't move them at the top of the `run` method, I've moved them at the top of the `class` definition in 25420dd374a398cdc2321a1a8dd1294ec22d481a, at least

Reply:
Ah, got it. Thank you!